utils/website_crawler.py

```diff
@@ -0,0 +1,54 @@
+from HTMLParser import HTMLParser
+import urllib
+
+
+class WebsiteCrawler(HTMLParser):
+    def __init__(self, raw_html, website_url):
+        HTMLParser.__init__(self)
+        print("Crawling possible websites...")
+        self.website_set = set()
+        self.website_url = website_url
+        self.raw_html = raw_html
+
+        self.website_set.add(website_url)
+
+    def handle_starttag(self, tag, attrs):
```

Comment:
Could try to refactor in separate methods.
Could make use of lambda expressions. 
Could also pre-process the file so you can get the <a> tags before you start the matching.